### DISCUSSION TITLE
Uptime logging

### DISCUSSION THREAD

**Discussion by rvry** [jasonacox/Powerwall-Dashboard#395] [General]
Hi Team,

I was wondering if gateway uptime is logged as part of the dashboard polls? I'd like to expose this into a graph as I've had some instability with my system and I suspect it starts around 30-40 days after the last gateway reset.

Some historical data on gateway reboots would help me with my hypothesis :-) My fallback is to look for historically logged alerts that usually fire when a gateway reset occurs.

Roger

**Comment by BJReplay** [comment]
POD_w051_SW_App_Boot Alert fires when the Gateway Boots.

Here's the last 30 days of alerts for me, including a Gateway Boot as part of a software upgrade.  You can see the boot.

![image](https://github.com/jasonacox/Powerwall-Dashboard/assets/37993507/9f72e0cc-db03-475f-b677-78ba36723d0a)

Here's that period zoomed in:

![image](https://github.com/jasonacox/Powerwall-Dashboard/assets/37993507/9925f009-fc84-4075-a9d3-d7e33a041010)

PodCommissionTime drops out while the Gateway is rebooting, so that's the alert I'd be looking for Gateway Uptime.

Tooltip over PodCommissionTime shows last reboot:

![image](https://github.com/jasonacox/Powerwall-Dashboard/assets/37993507/a232c655-0b81-453f-a78d-f07f5bee2541)

**Reply by rvry** [reply]
Very handy information, thank you so much! I'll see if I can expose that in a graph by evaluating the podCommissionTime alert and share it here :-)

**Comment by jasonacox** [comment]
The system does appear to track uptime in the /api/status URL which you can hit via the pypowerwall endpoint (e.g. http://localhost:8675/api/status - replace localhost with the address of your Dashboard host). It will provide a JSON payload like this:

```json
{
    "din": "1232100-00-E--TG123456789012",
    "start_time": "2023-10-13 04:01:45 +0800",
    "up_time_seconds": "1072h49m29.487426153s",
    "is_new": false,
    "version": "23.28.2 27626f98",
    "git_hash": "27626f98a66cad5c665bbe1d4d788cdb3e94fd33",
    "commission_count": 0,
    "device_type": "teg",
    "teg_type": "unknown",
    "sync_type": "v2.1",
    "cellular_disabled": false,
    "can_reboot": true
}
```
